rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLInsertOrUpdateExecutor.java

```diff
@@ -250,7 +250,7 @@ public TableRecords beforeImage() throws SQLException {
      */
     public TableRecords buildTableRecords2(TableMeta tableMeta, String selectSQL, ArrayList<List<Object>> paramAppenderList) throws SQLException {
         ResultSet rs = null;
-        try (PreparedStatement ps = statementProxy.getConnection().prepareStatement(selectSQL + " FOR UPDATE")) {
+        try (PreparedStatement ps = statementProxy.getConnection().prepareStatement(selectSQL)) {
```

Comment:
There is no way to ensure data consistency between beforeImage, execution, and afterImage.